### PR TITLE
Tempus: unit_test - add Dahlquist App Action test to DIRK

### PR DESCRIPTION
**Pull request by sconde**
Tempus: StepperIMEX_RK - reset stage number before solution computation

- to be consistent with all ERK/DIRK
- before the `AppAction::ACTION_LOCATION::END_STEP` call

Tempus: unit_test - IMEX_RK add testing for implicit stage time

@trilinos/tempus

### CODE REVIEW

**Review by ccober6 — APPROVED** [review]
Looks good!

**Comment by ccober6** [comment]
@sconde FYI, I get the same error when I build with "-Werror=unused-variable".

**Comment by sconde** [comment]
> @sconde FYI, I get the same error when I build with "-Werror=unused-variable".

Fixed.

File: packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp [hunk 1]
```diff
@@ -169,4 +173,205 @@ TEUCHOS_UNIT_TEST(IMEX_RK, AppAction)
...
+  auto stepper = rcp(new Tempus::StepperIMEX_RK<double>());
+  stepper->setModel(model);
+
+  // Default values for construction.
+  auto solver = rcp(new Thyra::NOXNonlinearSolver());
+  solver->setParameterList(Tempus::defaultSolverParameters());
+
+  bool useFSAL              = stepper->getUseFSAL();
+  std::string ICConsistency = stepper->getICConsistency();
+  bool ICConsistencyCheck   = stepper->getICConsistencyCheck();
```
Comment (ccober6) [inline packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp:336]: I still get an error on this ...
`/Users/ccober/myOffice/Trilinos_Second/packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp:336:8: error: unused variable 'ICConsistencyCheck' [-Werror,-Wunused-variable]
  bool ICConsistencyCheck   = stepper->getICConsistencyCheck();
       ^
1 error generated.
`

Comment (sconde) [reply packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp:336]: My apologies. I turned the warnings for my config script. I think I've caught them now

File: packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp [hunk 2]
```diff
@@ -169,4 +173,205 @@ TEUCHOS_UNIT_TEST(IMEX_RK, AppAction)
...
+  // Default construction.
+  auto stepper = rcp(new Tempus::StepperIMEX_RK<double>());
+  stepper->setModel(model);
+
+  // Default values for construction.
+  auto solver = rcp(new Thyra::NOXNonlinearSolver());
+  solver->setParameterList(Tempus::defaultSolverParameters());
+
+  bool useFSAL              = stepper->getUseFSAL();
+  std::string ICConsistency = stepper->getICConsistency();
```
Comment (ccober6) [inline packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp:335]: @sconde Oddly enough ICConsistency appears to be an unused variable, but it does not error out?!

**Review by ccober6 — DISMISSED** [review]
Looks good other than the odd non-error for ICConsistency.

File: packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp (hunk 2 above)
Comment (sconde) [reply packages/tempus/unit_test/Tempus_UnitTest_IMEX_RK.cpp:335]: Ok. I caught my embarrassing mistake. I was using a different config file that was not turning on the warning as errors for me. I've deleted my build directory, and rebuilt the PR and I'm not seeing any warnings from Tempus now.

**Review by ccober6 — APPROVED** [review]
Ok, looks good!

**Pull request closed without merging**